antiope-s3/src/Antiope/S3.hs

```diff
@@ -152,6 +154,21 @@ lsPrefix b p =
     .| mapC (S3Uri b . view oKey)
     .| sinkList
 
+-- | Lists the specified objects in a bucket, non-recursively.
+lsObjects :: MonadAWS m
+  => BucketName
+  -> ObjectKey
+  -> m [ObjectKey]
+lsObjects bkt dir =
+  let req = listObjectsV2 bkt & lovPrefix ?~ toText dir & lovDelimiter ?~ ('/' :: Delimiter)
+  in runConduit $ lsBucketResponseStream req .| CC.concatMap lsResponseToObjectsList .| consume
+  where
+    lsResponseToObjectsList rs = keys <> prefixes
```

Comment:
This looks odd. I may be misunderstanding what you're doing, but are you sure that `keys <> prefixes` is what you want to do here?

Reply:
Yes, just want to list all the entries under this directory, both files and sub directories, but non-recursively, for example:
With `keys <> prefixes`:
```
λ> runResourceT $ runAWS env $ lsObjects "lo-werphat--sinkhole-bins" "incoming/"
[ObjectKey "incoming/project.sh",ObjectKey "incoming/2019/"]
it :: [amazonka-s3-1.6.1:Network.AWS.S3.Internal.ObjectKey]
(0.88 secs, 2,635,008 bytes)
```
If without `keys`:
```
λ> runResourceT $ runAWS env $ lsObjects "lo-werphat--sinkhole-bins" "incoming/"
[ObjectKey "incoming/2019/"]
it :: [amazonka-s3-1.6.1:Network.AWS.S3.Internal.ObjectKey]
(1.20 secs, 2,597,624 bytes)
```

Reply:
Ah, I see. I mis-understood what those referred to.